spanner/row.go

```diff
@@ -378,3 +389,186 @@ func (r *Row) ToStructLenient(p interface{}) error {
 		true,
 	)
 }
+
+// SelectAll iterates all rows to the end. After iterating it closes the rows,
+// and propagates any errors that could pop up.
+// It expects that destination should be a slice. For each row it scans data and appends it to the destination slice.
+// SelectAll supports both types of slices: slice of structs by a pointer and slice of structs by value,
+// for example:
+//
+//	type Singer struct {
+//	    ID    string
+//	    Name  string
+//	}
+//
+//	var singersByPtr []*Singer
+//	var singersByValue []Singer
+//
+// Both singersByPtr and singersByValue are valid destinations for SelectAll function.
+//
+// Before starting, SelectAll resets the destination slice,
+// so if it's not empty it will overwrite all existing elements.
+func SelectAll(rows Iterator, v interface{}, options ...DecodeOptions) error {
+	if rows == nil {
+		return fmt.Errorf("rows is nil")
+	}
+	if v == nil {
+		return fmt.Errorf("p is nil")
+	}
+	dstVal := reflect.ValueOf(v)
+	if !dstVal.IsValid() || (dstVal.Kind() == reflect.Ptr && dstVal.IsNil()) {
+		return errNilSlicePointer()
+	}
+	if dstVal.Kind() != reflect.Ptr {
+		return errNotASlicePointer()
+	}
+	dstVal = dstVal.Elem()
+	dstType := dstVal.Type()
+	if k := dstType.Kind(); k != reflect.Slice {
+		return errNotASlicePointer()
+	}
+
+	itemType := dstType.Elem()
+	var itemByPtr bool
+	// If it's a slice of pointers to structs,
+	// we handle it the same way as it would be slice of struct by value
+	// and dereference pointers to values,
+	// because eventually we work with fields.
+	// But if it's a slice of primitive type e.g. or []string or []*string,
```

Comment:
I don't quite understand this. The documentation of the method says that the destination slice must be a slice of (pointers to) structs. This seems to indicate that we also support primitives.

Reply:
Yes we do support primitives, tests added.